### PR TITLE
hello_rust: changes for target support

### PR DESCRIPTION
**Pull request by ptka**
## Summary
Changes for target support (Raspberry PI Pico)

## Impact
nothing as it's new

## Testing
Simulation on MacBook Air M1 with Monterey 12.2.1 using rustc rustc 1.58.1 (stable) via Homebrew.
Raspberry PI Pico (RP2040)

### CODE REVIEW

**Comment by ptka** [comment]
Will be re-opened as the dependencies are fulfilled.

File: examples/hello_rust/hello_rust_main.rs [hunk 1]
```diff
@@ -33,6 +40,20 @@ extern "C"
     pub fn printf(format: *const u8, ...) -> i32;
 }
 
+/****************************************************************************
+ * Private functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Panic handler (needed for [no_std] compilation)
```
Comment (no1wudi) [inline examples/hello_rust/hello_rust_main.rs:48]: Should we keep the same comment style with NuttX ? Rust has its own comment rule for better intergration of codes and docs.

Comment (ptka) [reply examples/hello_rust/hello_rust_main.rs:48]: Definitely a good point. I followed the NuttX style to have a starting point, knowing, that this is most likely no there best solution.

May I ask you to provide more details, so we can go a step ahead on this point:

  - Is there a good official documentation on the Rust style, we can reference in NuttX?
  - Can we check this, similar to checkpatch.sh and can be used in CI?
  - Is the rust style compatible to current NuttX style, checked from checkpatch.sh?

Comment (no1wudi) [reply examples/hello_rust/hello_rust_main.rs:48]: > 
> May I ask you to provide more details, so we can go a step ahead on this point:
> 
> * Is there a good official documentation on the Rust style, we can reference in NuttX?

Please refer to https://doc.rust-lang.org/reference/comments.html, there are doc comments and non-doc comments.

> * Can we check this, similar to checkpatch.sh and can be used in CI?

Please refer https://lore.kernel.org/rust-for-linux/YO3VIjpaCSZnNapB@moria.home.lan/T, may rustfmtcheck will help you.

**Comment by xiaoxiang781216** [comment]
@ptka does the code pass rustfmt check?

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**